src/NUnitFramework/framework/Internal/Results/TestResult.cs

```diff
@@ -266,6 +266,11 @@ public virtual XmlNode AddToXml(XmlNode parentNode, bool recursive)
             thisNode.AddAttribute("start-time", StartTime.ToString("u"));
             thisNode.AddAttribute("end-time", EndTime.ToString("u"));
             thisNode.AddAttribute("duration", Duration.ToString("0.000000", NumberFormatInfo.InvariantInfo));
+            var testAssembly = TestUtility.GetTestAssembly(Test);
+            if (testAssembly != null)
+            {
+                thisNode.AddAttribute("assembly", testAssembly.FullName);
+            }
```

Comment:
I would prefer to avoid adding this attribute to the XML if possible. Every test is eventually nested under a suite of type "Assembly", which has the full name of the assembly. Putting it on every test is an extreme level of duplication in the result file for all users, even though it is only used for TeamCity.

Also, every test has an id, the first part of which (before the hyphen) identifies the assembly uniquely. It could be considered to use this as a part of the name, perhaps as a prefix, when creating TC messages.